uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/BAO/LeavePeriodEntitlementTest.php

```diff
@@ -755,14 +782,36 @@ public function testgetLeavePeriodEntitlementRemainder() {
     );
 
     $params = ['entitlement_id' => $periodEntitlement->id];
-    $result = $periodEntitlement::getLeavePeriodEntitlementRemainder($params);
+    $result = $periodEntitlement::getRemainder($params);
     $this->assertEquals(1, count($result));
     $this->assertEquals(7, $result[0]['remainder']['current']);
+    $this->assertEquals(7, $result[0]['remainder']['current']);
+    $this->assertArrayNotHasKey('future', $result[0]['remainder']);
   }
 
-  public function testgetLeavePeriodEntitlementRemainderWithIncludeFuture() {
-    $absence_period = $this->createAbsencePeriod(date('Y') .'-01-01', date('Y') .'-12-31');
-    $periodEntitlement = $this->createPeriodEntitlement($absence_period->id);
+  public function testGetMultipleLeavePeriodEntitlementRemainder() {
+    $absencePeriod = AbsencePeriodFabricator::fabricate([], true);
+    $periodEntitlement1 = LeavePeriodEntitlementFabricator::fabricate(['period_id' => $absencePeriod->id], true);
+
+    //create two more LeavePeriodEntitlement within same period with same contactid
+    $periodEntitlement2 = LeavePeriodEntitlementFabricator::fabricate(['period_id' => $absencePeriod->id, 'type_id' => 2], true);
+    $periodEntitlement3 = LeavePeriodEntitlementFabricator::fabricate(['period_id' => $absencePeriod->id, 'type_id' => 3], true);
+
+    $this->createLeaveBalanceChange($periodEntitlement1->id, 10);
+    $this->createLeaveBalanceChange($periodEntitlement2->id, 10);
+    $this->createLeaveBalanceChange($periodEntitlement3->id, 10);
```

Comment:
it would be nice to have different values for the entitlement, just to make sure that the returned values are not all the same